jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1033,6 +1033,27 @@ static <T> T as(Request request, Class<T> type)
         return null;
     }
 
+    /** Unwrap a Request back to the given type, ensuring that we do not cross a
+     * context boundary (as might be the case during cross-context dispatch).
+     * @param request the possibly wrapped request to unwrap
+     * @param type the type to unwrap to
+     * @param context the limiting context
+     * @return the request unwrapped back to the given type, or null if it cannot be
+     * unwrapped to that type or a context boundary is crossed.
+     */
```

Comment:
Add a @see as(Request,Class<T>) and can you give some javadoc for that method  too?